elastic/integrations
# synthetics - segment ilm policies by dataset

Type of change
Please label this PR with one of the following labels, depending on the scope of your change:
- Enhancement

## What does this PR do?

Resolves https://github.com/elastic/uptime/issues/462

Specifies ILM policy default for each synthetics data set, `synthetics-http`, `synthetics-tcp`, `synthetics-icmp`, `synthetics-browser.default`, `synthetics-browser.screenshot`, and `synthetics-browser.network`.

Specifically, it sets a default delete phase timeline for each data set. 

Below you can see the new ILM policies assigned to each data stream:

Browser
--
![image](https://user-images.githubusercontent.com/11356435/164538718-a60cefa5-0a04-4cb0-bbab-0bd94d372a5d.png)
![image](https://user-images.githubusercontent.com/11356435/164538530-a3ec8a36-9d25-4d95-859a-288b01237678.png)
![image](https://user-images.githubusercontent.com/11356435/164538622-49f5b18f-7d13-4d55-beb3-6989066e6818.png)

HTTP
--
![image](https://user-images.githubusercontent.com/11356435/164538839-54423629-af22-467c-9649-b8dc45f9a658.png)


TCP
--
![image](https://user-images.githubusercontent.com/11356435/164538475-a7980acb-47c3-45ed-9a2f-bb6b9fd50328.png)

ICMP
--
![image](https://user-images.githubusercontent.com/11356435/164538894-0c129de1-6aa5-4948-ac37-99e784837eda.png)

### Testing
1. Pull down this branch
2. Navigate in the repo directory to `/packages/synthetics`
3. Run `elastic-package clean`, then `elastic-package` build.
4. Run `elastic-package stack up --services "package-registry"`
5. Add the following keys to your local Kibana: https://p.elstc.co/paste/T6dpG-0P#cFd4gPpGiivr-6eQWKSyBN1UYgjfiZvPsd+iaYGFEIZ
6. Start Kibana locally. Navigate to Monitor Management. Create at least 4 monitors, one of each type
7. Navigate to Stack Management and click Data Streams
8. Confirm that you have synthetics data streams for each type
9. Click on each data stream. A sidebar should pop up. 
10. Confirm that the index lifecycle policy is unique to the data stream. The naming should be `synthetics-synthetics.[monitorType]-default_policy`
11. Click on the Index Lifecycle Policy linked to the data stream (you can see where to click on the screenshots).
12. This should take you to the index lifecycle page. Confirm that the Delete Phase is present, and that it matches the AC specified

|       | HTTP | TCP | ICMP | Browser default | Browser screenshot | Browser |
| --- | -----  | ---- | ----- | ---------------- | -------------------- | -------- |
| Delete timeframe (in days) | 365 | 365 | 365 | 365 | 14 | 14 |
